### PR TITLE
python3Packages.mkdocs-redoc-tag: init at 0.1.0

### PR DESCRIPTION
**Pull request by benhiemer**
## Description of changes

Add the mkdocs plugin mkdocs-redoc-tag.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/mkdocs-redoc-tag/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+, fetchFromGitHub
+, mkdocs
+, pytestCheckHook
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "mkdocs-redoc-tag";
+  version = "0.1.0";
+  format = "setuptools";
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-redoc-tag/default.nix:13]:
```suggestion
  pyproject = true;
```
and add `setuptools` to `nativeBuildInputs`.

File: maintainers/maintainer-list.nix
Comment (dotlambda) [inline maintainers/maintainer-list.nix:1]: Do this in a separate commit `maintainers: add benhiemer`.

File: pkgs/development/python-modules/mkdocs-redoc-tag/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    "test_material"
+    "test_material_dark_scheme_name"
+    "test_template"
+  ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-redoc-tag/default.nix:37]:
```suggestion
    mkdocs-material
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/mkdocs-redoc-tag/default.nix (hunk 1 above)
Comment (benhiemer) [reply pkgs/development/python-modules/mkdocs-redoc-tag/default.nix]: I implemented your suggestion. Thanks!

File: maintainers/maintainer-list.nix
Comment (benhiemer) [reply maintainers/maintainer-list.nix:1]: Done, thanks.
